### PR TITLE
Improving GuessIsDue method

### PR DESCRIPTION
**Pull request by ahmed605**

### CODE REVIEW

**Comment by janniksam** [comment]
@ahmed605 I did extract it into a new class and wrote a few tests for most cases. I took a look at the android app myself and yeah, it is indeed 100% the same logic.

File: Akinator.Api.Net/Utils/GuessDueChecker.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                return !(currentQuestion.Progression <= 97.0f);
+            }
+
+            if (currentQuestion.Progression <= 80.0f && stepsTakenSinceLastGuess < 30)
+            {
+                return false;
+            }
+
+            return 80 - currentQuestion.Step > 5;
+        }
```
Comment (ahmed605) [inline Akinator.Api.Net/Utils/GuessDueChecker.cs:81]: @janniksam I think the logic that I added in my latest commit was better that this (Android) one, I have tested it and got better results with the old (my latest commit) than this (Android) one

**Comment by ahmed605** [comment]
@janniksam BTW this is the logic that the Windows Phone app uses
WinPhone

**Comment by ahmed605** [comment]
@janniksam how about having an optional parameter in the GuessIsDue method (based on enum) which can set which logic the method will use (e.g `ANDROID`, `ANDROID_MODIFIED` and `WINDOWS_PHONE`) and I recommend that the default option be the `ANDROID_MODIFIED` logic that I added in my latest commit

**Comment by janniksam** [comment]
I would with just one of the implementations

File: Akinator.Api.Net/Utils/GuessDueChecker.cs (hunk 1 above)
Comment (janniksam) [reply Akinator.Api.Net/Utils/GuessDueChecker.cs:81]: I would like to keep it as original as possible. Hope you understand.

Comment (ahmed605) [reply Akinator.Api.Net/Utils/GuessDueChecker.cs:81]: ok, as you like :) it was just my opinion

**Comment by janniksam** [comment]
> @janniksam how about having an optional parameter in the GuessIsDue method (based on enum) which can set which logic the method will use (e.g `ANDROID`, `ANDROID_MODIFIED` and `WINDOWS_PHONE`) and I recommend that the default option be the `ANDROID_MODIFIED` logic that I added in my latest commit

Well, I rethought the whole situation. I think it might be a viable option.
But the parameter has to be optional, default should be the android implementation.

**Comment by ahmed605** [comment]
yeah, ok I already said that it will be optional :) 👍

File: Akinator.Api.Net/Enumerations/Platform.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Akinator.Api.Net.Enumerations
+{
+    public enum Platform
+    {
+        Android = 0,
+        Android_Modified = 1,
+        Windows_Phone = 3
```
Comment (janniksam) [inline Akinator.Api.Net/Enumerations/Platform.cs]: 0, 1, 3? :)

File: Akinator.Api.Net/IAkinatorClient.cs [hunk 3]
```diff
@@ -94,9 +94,8 @@ public interface IAkinatorClient : IDisposable
         /// Indicates, if Akinator is ready to guess based on a few criterias such as
         /// already answered questions or the probability given by Akinator
         /// </summary>
-        /// <param name="question">The last question asked by Akinator</param>
         /// <returns>Indicates whether Akinator is ready to guess or not</returns>
-        bool GuessIsDue(AkinatorQuestion question);
+        bool GuessIsDue(Platform platform = Platform.Android);
```
Comment (janniksam) [inline Akinator.Api.Net/IAkinatorClient.cs:99]: Can you add a description above?

File: Akinator.Api.Net/Utils/GuessDueChecker.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                            if (variance >= 10)
+                            {
+                                return true;
+                            }
+                        }
+                        return false;
+                    }
+                case Platform.Windows_Phone:
+                    {
+                        return currentQuestion.Step >= 79 || (currentQuestion.Step - lastGuessTakenAtStep >= 5 && (currentQuestion.Progression >= 97f || currentQuestion.Step - lastGuessTakenAtStep == 25) && currentQuestion.Step != 75);
```
Comment (janniksam) [inline Akinator.Api.Net/Utils/GuessDueChecker.cs:78]: I would extract the logics into 3 seperate methods.

File: Akinator.Api.Net/Enumerations/Platform.cs (hunk 2 above)
Comment (ahmed605) [reply Akinator.Api.Net/Enumerations/Platform.cs]: Oh, my bad sorry :joy:

File: Akinator.Api.Net/IAkinatorClient.cs (hunk 3 above)
Comment (ahmed605) [reply Akinator.Api.Net/IAkinatorClient.cs:99]: Sure, I will add

File: Akinator.Api.Net/Utils/GuessDueChecker.cs (hunk 4 above)
Comment (ahmed605) [reply Akinator.Api.Net/Utils/GuessDueChecker.cs:78]: I think it's better and more understandable like this

**Review by janniksam — APPROVED** [review]

**Pull request merged**